projects/injection/private/Injector.cxx

```diff
@@ -161,55 +171,57 @@ void Injector::SampleCrossSection(LI::dataclasses::InteractionRecord & record, s
+                        fake_record.signature = signature;
+                        fake_record.target_mass = detector_model->GetTargetMass(target);
+                        // Add total cross section times density to the total prob
+                        fake_prob = target_density * cross_section->TotalCrossSection(fake_record);
+                        total_prob += fake_prob;
+                        xsec_prob += fake_prob;
+                        // Add total prob to probs
+                        probs.push_back(total_prob);
+                        // Add target and cross section pointer to the lists
+                        matching_targets.push_back(target);
+                        matching_cross_sections.push_back(cross_section);
+                        matching_signatures.push_back(signature);
+                    }
+                }
+            }
+        }
     }
     if (interactions->HasDecays()) {
-      for(auto const & decay : interactions->GetDecays() ) {
-        for(auto const & signature : decay->GetPossibleSignaturesFromParent(record.signature.primary_type)) {
-          fake_record.signature = signature;
-          // fake_prob has units of 1/cm to match cross section probabilities
-          fake_prob = 1./(decay->TotalDecayLengthForFinalState(fake_record)/LI::utilities::Constants::cm);
-          total_prob += fake_prob;
-          // Add total prob to probs
-          probs.push_back(total_prob);
-          // Add target and decay pointer to the lists
-          matching_targets.push_back(LI::dataclasses::Particle::ParticleType::Decay);
-          matching_decays.push_back(decay);
-          matching_signatures.push_back(signature);
+        for(auto const & decay : interactions->GetDecays() ) {
+            for(auto const & signature : decay->GetPossibleSignaturesFromParent(record.signature.primary_type)) {
+                fake_record.signature = signature;
+                // fake_prob has units of 1/cm to match cross section probabilities
+                fake_prob = 1./(decay->TotalDecayLengthForFinalState(fake_record)/LI::utilities::Constants::cm);
+                total_prob += fake_prob;
+                // Add total prob to probs
+                probs.push_back(total_prob);
+                // Add target and decay pointer to the lists
+                matching_targets.push_back(LI::dataclasses::Particle::ParticleType::Decay);
+                matching_decays.push_back(decay);
+                matching_signatures.push_back(signature);
+            }
         }
-      }
     }
+
+    if(total_prob == 0)
+        throw(LI::utilities::InjectionFailure("No valid interactions for this event!"));
     // Throw a random number
     double r = random->Uniform(0, total_prob);
     // Choose the target and cross section
     unsigned int index = 0;
-    for(; (index < probs.size()-1) and (r > probs[index]); ++index) {}
+    for(; (index+1 < probs.size()) and (r > probs[index]); ++index) {}
```

Comment:
Yes, it does. The `probs.size()-1` will fail silently with an integer underflow if there are no valid cross sections or decays, since `size_t` is an unsigned integer.